Tests/LootLockerTests/PlayMode/GuestSessionTest.cs

```diff
@@ -0,0 +1,228 @@
+using System;
+using System.Collections;
+using System.Text.RegularExpressions;
+using LootLocker;
+using LootLocker.Requests;
+using LootLockerTestConfigurationUtils;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace LootLockerTests.PlayMode
+{
+    public class GuestSessionTest
+    {
+        private LootLockerTestGame gameUnderTest = null;
+        private LootLockerConfig configCopy = null;
+        private static int TestCounter = 0;
+        private bool SetupFailed = false;
+
+        [UnitySetUp]
+        public IEnumerator Setup()
+        {
+            TestCounter++;
+            configCopy = LootLockerConfig.current;
+
+            if (!LootLockerConfig.ClearSettings())
+            {
+                Debug.LogError("Could not clear LootLocker config");
+            }
+
+            // Create game
+            bool gameCreationCallCompleted = false;
+            LootLockerTestGame.CreateGame(testName: "GuestSessionTest" + TestCounter + " ", onComplete: (success, errorMessage, game) =>
+            {
+                if (!success)
+                {
+                    gameCreationCallCompleted = true;
+                    Debug.LogError(errorMessage);
+                    SetupFailed = true;
```

Comment:
This looks worrying, if ! success then set completed = true (outside code is now continuing), log the error, only then say we actually failed, and then no return meaning we also set the game under test to null (I presume) and then we set completed = true again?

Reply:
Fixed